libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -323,7 +323,7 @@ string IRGenerator::generateModifier(
 		t("functionName", functionName);
 		vector<string> retParamsIn;
 		for (auto const& varDecl: _function.returnParameters())
-			retParamsIn += IRVariable(*varDecl).stackSlots();
```

Comment:
Can you explain the problem here?

Reply:
I think the problem is that in line 366 of the same file, would end up calling `IRGeneratorForStatements::handleVariableReference`. And because the return variable was not in `m_context.m_localVariable`, it would throw.

Here is a stack trace:
```
#0  solidity::frontend::IRGeneratorForStatements::handleVariableReference (this=0x7fffffffc550, _variable=..., _referencingExpression=...)
    at /home/hari/s/r2/libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2353
#1  0x00000000008aa7df in solidity::frontend::IRGeneratorForStatements::endVisit (this=0x7fffffffc550, _identifier=...)
    at /home/hari/s/r2/libsolidity/codegen/ir/IRGeneratorForStatements.cpp:2278
#2  0x000000000070e58a in solidity::frontend::Identifier::accept (this=0x1300ef0, _visitor=...) at /home/hari/s/r2/libsolidity/ast/AST_accept.h:938
#3  0x00000000008849a3 in solidity::frontend::IRGeneratorForStatements::evaluateExpression (this=0x7fffffffc550, _expression=..., _targetType=...)
    at /home/hari/s/r2/libsolidity/codegen/ir/IRGeneratorForStatements.cpp:294
#4  0x0000000000868e2d in operator() (__closure=0x1370c40) at /home/hari/s/r2/libsolidity/codegen/ir/IRGenerator.cpp:369
#5  0x0000000000876729 in std::__invoke_impl<std::__cxx11::basic_string<char>, solidity::frontend::IRGenerator::generateModifier(const solidity::frontend::ModifierInvocation&, const solidity::frontend::FunctionDefinition&, const string&)::<lambda()>&>(std::__invoke_other, struct {...} &) (__f=...) at /usr/include/c++/10/bits/invoke.h:60
#6  0x0000000000875a15 in std::__invoke_r<std::__cxx11::basic_string<char>, solidity::frontend::IRGenerator::generateModifier(const solidity::frontend::ModifierInvocation&, const solidity::frontend::FunctionDefinition&, const string&)::<lambda()>&>(struct {...} &) (__fn=...) at /usr/include/c++/10/bits/invoke.h:115
#7  0x0000000000874dbc in std::_Function_handler<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >(), solidity::frontend::IRGenerator::generateModifier(const solidity::frontend::ModifierInvocation&, const solidity::frontend::FunctionDefinition&, const string&)::<lambda()> >::_M_invoke(const std::_Any_data &) (
    __functor=...) at /usr/include/c++/10/bits/std_function.h:292
#8  0x00000000007f1fce in std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > ()>::operator()() const (this=0x7fffffffcc10)
    at /usr/include/c++/10/bits/std_function.h:622
#9  0x00000000007f1b1a in solidity::frontend::MultiUseYulFunctionCollector::createFunction(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::function<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > ()> const&) (this=0x7fffffffd988, _name="modifier_mod2_6", _creator=...)
    at /home/hari/s/r2/libsolidity/codegen/MultiUseYulFunctionCollector.cpp:50
#10 0x00000000008696cf in solidity::frontend::IRGenerator::generateModifier (this=0x7fffffffd860, _modifierInvocation=..., _function=..., _nextFunction="fun_f_11_inner")
    at /home/hari/s/r2/libsolidity/codegen/ir/IRGenerator.cpp:386
```

Reply:
It's not a problem that it is added multiple times?

Reply:
Ah ok, now I see. This might be fine :)